### PR TITLE
[Model Update]: ItemStock

### LINKED ISSUE #310: [Model Update]: Make ProductStock an ItemStock
## Update Reason
In PURIS, the ProductStock aspect was meant for supplier side stocks. It should be standardized to be also applicable to the customer side. One key problem were legal concerns regarding multi-sourcing.

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists
- [x] The potential updates are discussed with all stakeholders
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [x] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description
The model has been renamed to item stock to cover stock on supplier and customer side.
It incorporates the shared aspects for uuid and unit of measure.

Closes #310 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by agg3fe** [comment]
@tom-rm-meyer-ISST please update the PR.

**Comment by tom-rm-meyer-ISST** [comment]
@agg3fe merged main.

File: io.catenax.item_stock/2.0.0/ItemStock.ttl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+] :lastUpdatedOnDateTime :allocatedStocks).
+:orderPositionReference a samm:Property;
+    samm:preferredName "Order Position Reference"@en;
+    samm:description "The order position reference contains information to identify a position within an order."@en;
+    samm:characteristic :OrderPositionReferenceCharacteristic.
+:lastUpdatedOnDateTime a samm:Property;
+    samm:preferredName "Lasted Updated on Date"@en;
+    samm:description "Date and time at which the position has been updated by the supplier's systems. Date and time are set according to a timezone."@en;
+    samm:characteristic :DateTimeStamp;
+    samm:exampleValue "2023-04-01T14:23:00"^^xsd:date.
```
Comment (agg3fe) [inline io.catenax.item_stock/2.0.0/ItemStock.ttl:79]: The Characteristic used is xsd:date. Either you need to change the example value to only date or change the Characteristic.

File: io.catenax.item_stock/2.0.0/ItemStock.ttl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+:isBlocked a samm:Property;
+    samm:preferredName "Is Blocked"@en;
+    samm:description "A set of finished items that are located on the blocked stock area, quarantine area, etc. waiting to be released.\n\nExample: Stock is in review and not released due to customer complaint regarding a recent batch."@en;
+    samm:characteristic :IsBlockedCharacteristic;
+    samm:exampleValue "false"^^xsd:boolean.
+:stockLocationBPNA a samm:Property;
+    samm:preferredName "BPNA of Stock Location"@en;
+    samm:description "BPNA identifying the address the stock is located at. One stock has always one location."@en;
+    samm:characteristic ext-number:BpnaTrait;
+    samm:exampleValue "BPNA1234567890";
```
Comment (agg3fe) [inline io.catenax.item_stock/2.0.0/ItemStock.ttl:142]: invalid example value. perhaps you can change it to something like BPNA1234567890ZZ

**Review by agg3fe — APPROVED** [review]
LGTM

**Pull request merged**